dojo/tools/whitehat_sentinel/parser.py

```diff
@@ -0,0 +1,244 @@
+        """
+
+        endpoints_list = []
+
+        # This should be in the Endpoint class should it not?
+        for attack_vector in attack_vectors:
+            try:
+                url = attack_vector['request']['url']
+                parsed_url = urlparse(url)
+                protocol = parsed_url.scheme
+                query = parsed_url.query
+                fragment = parsed_url.fragment
+                path = parsed_url.path
+                port = ""
+                try:
+                    host, port = parsed_url.netloc.split(':')
+                except ValueError:
+                    host = parsed_url.netloc
+
+                endpoints_list.append(Endpoint(host=host,
+                                               port=port,
+                                               path=path,
+                                               protocol=protocol,
+                                               query=query,
+                                               fragment=fragment)
+                                      )
+            except Exception:
+                pass
+
+        return endpoints_list
+
+    def _convert_whitehat_sentinel_vuln_to_dojo_finding(self, whitehat_sentinel_vuln, test):
+        """
+        Converts a WhiteHat Sentinel vuln to a DefectDojo finding
+
+        Args:
+            whitehat_sentinel_vuln: The vuln dictionary from WhiteHat Sentinel vuln API
+            test: The test ID that the DefectDojo finding should be associated with
+        Returns: A DefectDojo Finding object
+        """
+
+        # Out of scope is considered Active because the issue is valid, just not for the asset in question. If the
+        # issue belongs to another Product, it should be filtered before importing.
+        active = whitehat_sentinel_vuln.get('status') in ('open', 'out of scope')
+        date_created = whitehat_sentinel_vuln['found'].split('T')[0]
+        mitigated_ts = whitehat_sentinel_vuln.get('closed'.split('T')[0], None)
+        cwe = self._parse_cwe_from_tags(whitehat_sentinel_vuln['attack_vectors'][0]['scanner_tags'])
+        description_ref = self._parse_description(whitehat_sentinel_vuln['description'])
+        description = description_ref['description']
+        references = description_ref['reference_link']
+        steps = whitehat_sentinel_vuln['description'].get('description_prepend', '')
+        solution = self._parse_solution(whitehat_sentinel_vuln['solution'])
+        risk_id = whitehat_sentinel_vuln.get('custom_risk') if whitehat_sentinel_vuln.get(
+            'custom_risk') else whitehat_sentinel_vuln.get('risk')
+        severity = self._convert_whitehat_severity_id_to_dojo_severity(risk_id)
+        false_positive = whitehat_sentinel_vuln.get('status') == 'invalid'
+
+        finding = Finding(title=whitehat_sentinel_vuln['class'],
+                          test=test,
+                          cwe=cwe,
+                          active=active,
```

Comment:
`active` need to be aligned with mitigation. (if mitigated, active is false)

Reply:
done!